### PR TITLE
NOTICK: Tighten-up regex checks

### PR DESCRIPTION
**Pull request by vkolomeyko**
Avoid using `.*` wildcard to avoid injection of `/` to allow navigating sub-resources/commands

### CODE REVIEW

**Comment by vkolomeyko** [comment]
@mnesbit - as discussed yesterday. Changing wildcards to prevent traversal on sub-resources using `/`.

File: tools/plugins/initial-rbac/src/main/kotlin/net/corda/cli/plugin/initialRbac/commands/RoleCreationUtils.kt [hunk 1]
```diff
@@ -16,6 +16,13 @@ import java.time.temporal.ChronoUnit
 
 internal object RoleCreationUtils {
 
+    private const val UUID_CHARS = "[a-fA-F0-9]"
+    const val UUID_REGEX = "$UUID_CHARS{8}-$UUID_CHARS{4}-$UUID_CHARS{4}-$UUID_CHARS{4}-$UUID_CHARS{12}"
+
+    const val VNODE_SHORT_HASH_REGEX = "$UUID_CHARS{12}"
+
+    const val USER_REGEX = "[-._@a-zA-Z0-9]{3,250}"
+
```
Comment (benmcmahon100) [inline tools/plugins/initial-rbac/src/main/kotlin/net/corda/cli/plugin/initialRbac/commands/RoleCreationUtils.kt]: All look conceptually good. It'd be nice to accompany these with comments providing a sample of what they should match so they're easy to check at a glance

Comment (vkolomeyko) [reply tools/plugins/initial-rbac/src/main/kotlin/net/corda/cli/plugin/initialRbac/commands/RoleCreationUtils.kt]: Agree - I will probably add some unit tests to prove that it works as expected.
With RegEx it is too easy to make a mistake.

- [x] Add unit tests

Comment (sprintengine) [reply tools/plugins/initial-rbac/src/main/kotlin/net/corda/cli/plugin/initialRbac/commands/RoleCreationUtils.kt]: I think user login names are 255 chars long in the database, should USER_REGEX be up to 255 chars?

**Review by benmcmahon100 — APPROVED** [review]
LGTM

**Review by conal-smith — APPROVED** [review]
LGTM

**Review by pmck91 — APPROVED** [review]
LGTM

**Pull request merged**